dolphinscheduler-service/src/main/java/org/apache/dolphinscheduler/service/process/TriggerRelationServiceImpl.java

```diff
@@ -45,29 +51,44 @@ public void saveTriggerToDb(ApiTriggerType type, Long triggerCode, Integer jobId
         triggerRelation.setUpdateTime(new Date());
         triggerRelationMapper.upsert(triggerRelation);
     }
+
     @Override
-    public TriggerRelation queryByTypeAndJobId(ApiTriggerType apiTriggerType, int jobId) {
+    public List<TriggerRelation> queryByTypeAndJobId(ApiTriggerType apiTriggerType, int jobId) {
         return triggerRelationMapper.queryByTypeAndJobId(apiTriggerType.getCode(), jobId);
     }
 
     @Override
     public int saveCommandTrigger(Integer commandId, Integer processInstanceId) {
-        TriggerRelation exist = queryByTypeAndJobId(ApiTriggerType.PROCESS, processInstanceId);
-        if (exist == null) {
+        List<TriggerRelation> existTriggers = queryByTypeAndJobId(ApiTriggerType.PROCESS, processInstanceId);
+        if (CollectionUtils.isEmpty(existTriggers)) {
             return 0;
         }
-        saveTriggerToDb(ApiTriggerType.COMMAND, exist.getTriggerCode(), commandId);
-        return 1;
+        existTriggers.forEach(triggerRelation -> saveTriggerToDb(ApiTriggerType.COMMAND,
```

Comment:
If returning more than 1 result is a situation that should not occur, are there any potential problems if they are still stored into db?

Reply:
AFAIK, this table is only used for querying the workflow instance by rest-api, it will to affect the other api.

BTW, this table can be removed if we can generate the workflow instance in api-side when manual trigger a workflow instance, I have a plan to create a DSIP for this. This can help to reduce the complexity.